src/bp/lang-server/service/index.ts

```diff
@@ -19,6 +25,14 @@ const FAST_TEXT_MODEL_REGEX = /^(\w+)\.(\w+)\.(\d+)\.bin$/i
 // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
 const BPE_MODEL_REGEX = /^(\w+)\.(\w+)\.bpe\.model$/i
 
+const checkEnoughRam = (langNb: number, dims: number): RamInfos => {
+  return {
+    used: Number.parseFloat((os.freemem() / (1024 * 1024 * 1024)).toFixed(2)),
+    total: Number.parseFloat((os.totalmem() / (1024 * 1024 * 1024)).toFixed(2)),
+    prediction: (dims > 101 ? 4.0 : 2.0) * langNb
```

Comment:
Extract `4.0` and `2.0` in constant. Something like `GB_PER_LANG`.

Reply:
I added a fonction based on a linear regression to get the size (more generic).